### PR TITLE
Default packaging without OCAML backend [brew-staging: remove-opam]

### LINKED ISSUE #1600: ocaml-less packages
Could we have an ocaml-free K package (maybe called kframework-lite_5.0.0_amd64_focal.deb ), so that users who do not need the ocaml backend do not have to wait for ocaml build?

### PR DESCRIPTION
**Pull request by ehildenb**
Fixes: #1600 

This PR does:

-   Removes the OCAML backend dependency setup from the package installation process.
-   Removes the OCAML backend dependency teardown from the package removal process.
-   Updates the INSTALL and README instructions to let people know that OCAML dependencies are **OPTIONAL** and how to go about setting them up.

### CODE REVIEW

**Comment by ehildenb** [comment]
Base branch for homebrew-k repo: https://github.com/kframework/homebrew-k/compare/remove-opam

File: Jenkinsfile [hunk 1]
```diff
@@ -103,6 +103,7 @@ pipeline {
                   options { skipDefaultCheckout() }
                   steps {
                     unstash 'bionic'
+                    sudo apt-get install --yes opam
                     sh 'src/main/scripts/test-in-container-debian'
```
Comment (sskeirik) [inline Jenkinsfile]:
```suggestion
                    sh '''
                      sudo apt-get install --yes opam
                      src/main/scripts/test-in-container-debian
                    '''
```

File: src/main/scripts/test-in-container [hunk 2]
```diff
@@ -1,4 +1,9 @@
 #!/bin/sh -ex
+k-configure-opam-dev || true
+opam depext conf-pkg-config.1.1 || true
+eval $(opam config env)
+opam depext conf-pkg-config.1.1 || true
+false
```
Comment (sskeirik) [inline src/main/scripts/test-in-container]:
```suggestion
opam depext conf-pkg-config.1.1
k-configure-opam-dev
eval $(opam config env)
```

**Comment by ehildenb** [comment]
@dwightguth this is ready.

-  Evidence it works on Debian, Focal, Bionic, and MacOS: https://office.runtimeverification.com/jenkins/blue/organizations/jenkins/k/detail/PR-1618/24/pipeline
-  Evidence that it works on Arch: https://office.runtimeverification.com/jenkins/blue/organizations/jenkins/k/detail/PR-1618/25/pipeline

**Review by dwightguth — APPROVED** [review]

**Pull request merged**